docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+be made here. How exactly does the database intend to alter the ``User`` 
+type if we say yes? Simply clicking ``l`` will show it:
+
+.. code-block::
+
+    The following DDL statements will be applied:
+      ALTER TYPE default::User {
+          CREATE PROPERTY nam: std::str;
+          DROP PROPERTY name;
+      };
+
+This shows us clear as day that saying ``yes`` will result in creating 
+a new property called ``nam`` and dropping the existing ``name`` property.
+
+So when doubts dwell, click the letter l!
+
+``c`` (or ``confirmed``)
+------------------------
+
+This simply shows the entire list of statements that have been confirmed. 
+In other words, this is the migration as it stands at this point.
+
+``b`` (or ``back``)
+-------------------
+
+This will undo the last confirmation you agreed to and move you back 
+a step in the migration. If you haven't confirmed any statements yet, 
+a message will simply appear to let you know that there is nowhere 
+further back to move to. So pressing ``b`` will never abort a migration.
+
+The following two keys will stop the migration, but in different ways:
+
+``s`` (or ``stop``)
+-------------------
+
+This is also known as a 'split'. Pressing ``s`` will complete the 
+migration at the current point. Any statements that you have applied 
+will be applied, but the schema will not yet match the schema in your 
+``.esdl`` file(s). But you can easily start another migration to complete 
+the remaining changes once you have applied the migration that was 
+just created. This effectively splits the migration into two or more 
+files.
+
+``q`` (or ``quit``)
+-------------------
+
+Pressing ``q`` will simply quit without saving any of your progress.
+
+Migration hashes and data migrations
+====================================
+
+Sometimes you may want to initialize a database with some default 
+data, or add some data to a migration that you have just created before 
+you apply it.
+
+EdgeDB assumes by default that a migration involves a change to your 
+schema, so it won't create a migration for you if it doesn't see a 
+schema change:
+
+.. code-block:: bash
+
+    $ edgedb migration create
+    No schema changes detected.
+
+So how do you create a migration with only data? To do this, just 
+add ``--allow-empty`` to the command:
+
+.. code-block:: bash
+
+    $ edgedb migration create --allow-empty
+    Created myproject/dbschema/migrations/00002.edgeql,
+    id: m1xseswmheqzxutr55cu66ko4oracannpddujg7gkna2zsjpqm2g3a
+
+You will now see an empty migration in which you can enter some queries. 
+It will look something like this:
+
+.. code-block::
+
+    CREATE MIGRATION m1xseswmheqzxutr55cu66ko4oracannpddujg7gkna2zsjpqm2g3a
+        ONTO m1n5lfw7n74626cverbjwdhcafnhmbezjhwec2rbt46gh3ztoo7mqa
+    {
+    };
+
+You can now add your queries inside the braces. Assuming a schema 
+with a simple ``User`` type, we could then add a bunch of queries 
```

Comment:
Rather than showing this way to add the data migrations and manually update the hash, should we demonstrate [`edgedb migration edit`](https://www.edgedb.com/docs/cli/edgedb_migration/edgedb_migration_edit) which will open the migration in your editor and automatically update the hash for you? Then we could circle back and say that, if you had edited the migration directly instead, you would have gotten this error and could manually update the hash.

Reply:
Good idea. Interestingly I recently added migration edit to Windows but didn't know that it also updated the hash.

https://github.com/edgedb/edgedb/commit/e7ff59dbf5b5eed95cb4600c158b0bb8eac35a4e

I've opted for a different order though as I want to walk the reader through the process first before showing the more convenient way to do it. Plus, even though `edgedb migration edit` is easier to *use*, it's still an extra item to remember when getting used to migrations in the first place whereas everyone knows how to edit a file, click save and follow the instructions in the output. So I'd still call it the more advanced concept and put it second.